### PR TITLE
[INLONG-7792][Manager] Support export Excel template file of StreamSource

### LINKED ISSUE #7792: [Feature][Manager] Support export Excel template file of StreamSource
### Description

Excel is one of the common office software formats for most enterprises. For InLong, it is very convenient to import and export batch information, such as batch operations of table fields, by supporting direct parsing of files.

### PR DESCRIPTION
**Pull request by featzhang**
## [INLONG-7792][Manager] Support parsing and exporting Excel files

Excel is one of the common office software formats for most enterprises. For InLong, it is very convenient to import and export batch information, such as batch operations of table fields, by supporting direct parsing of files.

Considering that there will be scenarios of importing and exporting Excel in the future, here we have only made a thin encapsulation on Apache POI to make it as universal as possible.

- Fixes #7792 

### Validation Results

image

### CODE REVIEW

**Review by healchow — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelCellDataTransfer.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        Object parseFromText(String so) {
+            if (so != null && so.length() > 0) {
+                String s = so;
+                Date date = null;
+
+                try {
+                    date = this.simpleDateFormat.parse(s);
+                } catch (ParseException var5) {
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelCellDataTransfer.java:48]: Why is it named var5？

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelCellDataTransfer.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                Date date = (Date) o;
+                return this.simpleDateFormat.format(date);
+            } else {
+                return String.valueOf(o);
+            }
+        }
+
+        Object parseFromText(String so) {
+            if (so != null && so.length() > 0) {
+                String s = so;
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelCellDataTransfer.java:43]: Why is it named so？

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+                    "At least one field must be marked as Excel Field by annotation @ExcelField in class " + clazz);
+        }
+        XSSFWorkbook hwb = new XSSFWorkbook();
+        XSSFSheet sheet = hwb.createSheet("Sheet 1");
+
+        for (int index = 0; index < heads.size(); index++) {
+            sheet.setColumnWidth(index, DEFAULT_COLUMN_WIDTH);
+        }
+        fillSheetHeader(sheet.createRow(0), heads);
+        //
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java:117]: Remove it.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+        hwb.write(out);
+        out.close();
+        LOGGER.info("Database export succeeded");
+    }
+
+    /**
+     * Fills the output stream with the provided class meta.
+     */
+    public static <T> void write(Class<T> clazz, OutputStream out) throws IOException {
+        Preconditions.expectTrue(clazz != null, "Class can not be empty!");
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java:131]: Preconditions.expectNotNull

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+    private static void fillSheetContent(XSSFSheet sheet, List<String> heads, List<Map<String, String>> contents) {
+        Optional.ofNullable(contents)
+                .ifPresent(c -> IntStream.range(0, c.size())
+                        .forEach(lineId -> {
+                            Map<String, String> line = contents.get(lineId);
+                            Row row = sheet.createRow(lineId + 1);
+                            IntStream.range(0, heads.size()).forEach(colId -> {
+                                String title = heads.get(colId);
+                                String ov = line.get(title);
+                                String value = ov == null ? "" : ov;
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java:148]: StringUtils.isBlank(ov).  Why is it named ov?

**Review by fuweng11 — APPROVED** [review]

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelCellDataTransfer.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            } else if (cellValue instanceof Date) {
+                Date date = (Date) cellValue;
+                return this.simpleDateFormat.format(date);
+            } else {
+                return String.valueOf(cellValue);
+            }
+        }
+
+        Object parseFromText(String cellValue) {
+            if (cellValue != null && cellValue.length() > 0) {
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelCellDataTransfer.java:45]: Use StringUtils.isNotBlank

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java [hunk 7]
```diff
@@ -0,0 +1,351 @@
...
+    private static void fillSheetContent(XSSFSheet sheet, List<String> heads, List<Map<String, String>> contents) {
+        Optional.ofNullable(contents)
+                .ifPresent(c -> IntStream.range(0, c.size())
+                        .forEach(lineId -> {
+                            Map<String, String> line = contents.get(lineId);
+                            Row row = sheet.createRow(lineId + 1);
+                            IntStream.range(0, heads.size()).forEach(colId -> {
+                                String title = heads.get(colId);
+                                String originValue = line.get(title);
+                                String value = originValue == null ? "" : originValue;
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java:148]: Use StringUtils.isNotBlank.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java [hunk 8]
```diff
@@ -0,0 +1,351 @@
...
+
+    /**
+     * Convert a list of objects to a list of maps, where each map represents an object's fields and values.
+     *
+     * @param objects The list of objects to be converted.
+     * @param <E>     The type of the objects to be converted.
+     * @return A list of maps, where each map represents an object's fields and values.
+     */
+    public static <E> List<Map<String, String>> write2List(List<E> objects) {
+        E e1 = objects.get(0);
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java:170]: Why is it named e1

**Review by fuweng11 — APPROVED** [review]

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongStreamController.java [hunk 9]
```diff
@@ -183,4 +191,22 @@ public Response<List<StreamField>> parseFields(@RequestBody ParseFieldRequest pa
         return Response.success(streamService.parseFields(parseFieldRequest));
     }
 
+    @RequestMapping(value = "/stream/fieldsImportTemplate", method = RequestMethod.GET)
+    @ApiOperation(value = "Download fields import template")
+    public void downloadFieldsImportTemplate(HttpServletResponse response) {
```
Comment (healchow) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongStreamController.java:196]: Suggest return true of false for the API method, you can wrap it with `Response`.

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongStreamController.java [hunk 10]
```diff
@@ -183,4 +191,22 @@ public Response<List<StreamField>> parseFields(@RequestBody ParseFieldRequest pa
...
+        response.setHeader("Content-Disposition",
+                "attachment;filename=" + fileName);
+        response.setContentType("multipart/form-data");
+
+        try {
+            ServletOutputStream outputStream = response.getOutputStream();
+            ExcelTool.write(StreamField.class, outputStream);
+
+        } catch (IOException e) {
+            log.error("Can not properly download Excel file", e);
```
Comment (healchow) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongStreamController.java:208]: If error, please throw an `BusinessException`.

**Review by healchow — CHANGES_REQUESTED** [review]

**Review by healchow — APPROVED** [review]

**Pull request merged**